lib/scala/kg/src/main/scala/stores/kg/mem/MemKgStore.scala

```diff
@@ -86,6 +86,12 @@ class MemKgStore extends KgStore {
   override def getRandomNode: KgNode =
     nodes(random.nextInt(nodes.size))
 
+  final override def getTopEdgesByObject(limit: Int, objectNodeId: String): List[KgEdge] =
+    edges.filter(_.`object` == objectNodeId).groupBy(_.predicate).mapValues(_.sortBy(edge => nodesById(edge.subject).pageRank.get)(Ordering[Double].reverse).take(limit)).values.flatten.toList
```

Comment:
Nice looking functional code here. You assume the pageRank exists, but I guess that's OK.

Reply:
We don't use get*EdgesByObject, as I mentioned in the comment. I'd either remove it or refactor this code to take the filter function e.g., getTopEdges(filter: (KgEdge) => Boolean), so both API methods can use it.